packages/matter.js/src/net/Network.ts

```diff
@@ -9,14 +9,60 @@ import { UdpChannel, UdpChannelOptions } from "./UdpChannel.js";
 
 export class NetworkError extends MatterError {}
 
+/**
+ * @see {@link MatterSpecification.v11.Core} § 11.11.4.4
+ * Duplicated from the GeneralDiagnostics cluster to avoid circular dependencies.
+ */
+export enum InterfaceType {
+    /**
+     * Indicates an interface of an unspecified type.
+     */
+    Unspecified = 0,
+
+    /**
+     * Indicates a Wi-Fi interface.
+     */
+    WiFi = 1,
+
+    /**
+     * Indicates a Ethernet interface.
+     */
+    Ethernet = 2,
+
+    /**
+     * Indicates a Cellular interface.
+     */
+    Cellular = 3,
+
+    /**
+     * Indicates a Thread interface.
+     */
+    Thread = 4,
+}
+
+export type NetworkInterface = {
+    name: string;
+    type?: InterfaceType;
+};
+
+export type NetworkInterfaceDetails = {
+    mac: string;
+    ipV4: string[];
+    ipV6: string[];
+};
+
+export type NetworkInterfaceDetailed = NetworkInterface & NetworkInterfaceDetails;
+
+export type NetworkInterfaceNameToTypeMapping = { name: string; type: InterfaceType }[];
```

Comment:
Rather than:

```
 MATTER_NETWORK_INTERFACENAMETOTYPEMAPPING_ETH0=1
```

Could we do:

```
MATTER_NETWORK_INTERFACE_ETH0_TYPE=ethernet
```

Thinking being if we ever want to add additional per-interface config properties then there is room to add here without creating an additional mapping object.

Also in general I've tried to use strings rather than numbers for config values just to make things less cryptic in contexts where the enum definition isn't available (e.g. env var/config file/command line).

Reply:
Ok cool. Did not knew the parser can do that fancy stuff. I was more thinking about this is set programmatically. Is that then array with {type:"..."} ??

Reply:
If you did `vars.get("network.interface")` it should give you the config subtree `{ eth0: { type: "ethernet"}}`

So it'd be a `Record<string, { type: string }>` though of course the type would need validation.  I figure at some point we could validate config with Matter schema 🙂

Reply:
Will verify

Reply:
@lauckhart how about this?